components/properties/src/props.rs

```diff
@@ -35,216 +35,146 @@ pub enum EnumeratedProperty {
     ConnectorPunctuation = 22,
-    /// An initial quotation mark
+    /// (`Pi`) An initial quotation mark
     InitialPunctuation = 28,
-    /// A final quotation mark
+    /// (`Pf`) A final quotation mark
     FinalPunctuation = 29,
-    /// A punctuation mark of other type
+    /// (`Po`) A punctuation mark of other type
     OtherPunctuation = 23,
 
-    /// A symbol of mathematical use
+    /// (`Sm`) A symbol of mathematical use
     MathSymbol = 24,
-    /// A currency sign
+    /// (`Sc`) A currency sign
     CurrencySymbol = 25,
-    /// A non-letterlike modifier symbol
+    /// (`Sk`) A non-letterlike modifier symbol
     ModifierSymbol = 26,
-    /// A symbol of other type
+    /// (`So`) A symbol of other type
     OtherSymbol = 27,
 }
 
-/// Enumerated property General_Category.
+/// Groupings of multiple General_Category property values.
 ///
-/// General_Category specifies the most general classification of a code point, usually
-/// determined based on the primary characteristic of the assigned character. For example, is the
-/// character a letter, a mark, a number, punctuation, or a symbol, and if so, of what type?
+/// Instances of `GeneralCategoryGroup` represent the defined multi-category
+/// values that are useful for users in certain contexts, such as regex. In
+/// other words, unlike [`GeneralCategory`], this supports groups of general
+/// categories: for example, `Letter` /// is the union of `UppercaseLetter`,
+/// `LowercaseLetter`, etc.
+///
+/// See <https://www.unicode.org/reports/tr44/> .
 ///
 /// The discriminants correspond to the `U_GC_XX_MASK` constants in ICU4C.
-/// Unlike [`GeneralSubcategory`], this supports groups of general categories: for example, `Letter`
+/// Unlike [`GeneralCategory`], this supports groups of general categories: for example, `Letter`
 /// is the union of `UppercaseLetter`, `LowercaseLetter`, etc.
 ///
-/// See <https://www.unicode.org/reports/tr44/> .
-///
 /// See `UCharCategory` and `U_GET_GC_MASK` in ICU4C.
 #[derive(Copy, Clone, PartialEq, Debug, Eq)]
 #[repr(transparent)]
-pub struct GeneralCategory(pub(crate) u32);
+pub struct GeneralCategoryGroup(pub(crate) u32);
 
 use GeneralCategory as GC;
-use GeneralSubcategory as GS;
+use GeneralCategoryGroup as GCG;
 
 #[allow(non_upper_case_globals)]
-impl GeneralCategory {
-    /// (`Lu`) An uppercase letter
-    pub const UppercaseLetter: GeneralCategory = GC(1 << (GS::UppercaseLetter as u32));
-    /// (`Ll`) A lowercase letter
-    pub const LowercaseLetter: GeneralCategory = GC(1 << (GS::LowercaseLetter as u32));
-    /// (`Lt`) A digraphic letter, with first part uppercase
-    pub const TitlecaseLetter: GeneralCategory = GC(1 << (GS::TitlecaseLetter as u32));
-    /// (`Lm`) A modifier letter
-    pub const ModifierLetter: GeneralCategory = GC(1 << (GS::ModifierLetter as u32));
-    /// (`Lo`) Other letters, including syllables and ideographs
-    pub const OtherLetter: GeneralCategory = GC(1 << (GS::OtherLetter as u32));
```

Comment:
Whoa, no, I don't think this is a move in the right direction. 

The purpose of the enum this patch renames to GeneralCategoryGroup was to make it possible for the user to query for anything in the General_Category table without having to worry about the specific details of ICU's model. If you are not steeped in this stuff, it's not obvious whether something is a top-level group or not. Requiring users to use a different API depending on details that aren't relevant to them just adds friction. I think this change moves in the direction of increased conceptual clarify for *us* (as implementers of ICU4X) at the cost of a less ergonomic API for users. I don't think that's a good trade-off. Like ICU4C, our API should let users ask "is this a letter?" or "is this an uppercase letter?" using the same function; you shouldn't have to be a Unicode expert to use ICU4X.

My previous understanding was that, after discussing this several times, we had come to a consensus on the existing design. I approved the linked issue because I thought we were just bikeshedding names, not changing functionality. If there are other benefits to making this change, I would appreciate a more detailed write-up of the justification.

Reply:
I have no strong opinions on this and lean toward @iainireland's reasoning above.

Reply:
The new enum names are good, but I agree with Iain that it's useful to retain the single-element groups.

The API docs still say “The discriminants correspond to the `U_GC_XX_MASK` constants in ICU4C.” and ICU4C does have those single-element constants.